pretyping/coercion.ml

```diff
@@ -643,7 +650,8 @@ let inh_coerce_to_fail ?(use_coercions=true) flags env sigma rigidonly v v_ty ta
           let sigma = unify_leq_delay ~flags env sigma direct_v rev_x in
           (try let _ = Evarutil.head_evar sigma (whd_evar sigma x) in raise Not_found
            with NoHeadEvar -> ());  (* fail if x is stil an unresolved evar *)
-          sigma, x, target_type, ReplaceCoe x
+          let sigma, rev_x = add_rev_coerce env sigma target_type v_ty x v in
+          sigma, rev_x, target_type, ReplaceCoe x
```

Comment:
Perhaps ReplaceCoe could be extended so that reapply_coercions also inserts the rev_coerce?